### PR TITLE
Update openid.md

### PR DESCRIPTION
**Pull request by andrewperry**
Providing more instructions around generic OIDC configuration, especially the setting of the user_profile_method

### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#changelog). The entry should:
  - Be a short description of your change which makes sense to users. "Fixed a bug that prevented receiving messages from other servers." instead of "Moved X method from `EventStore` to `EventWorkerStore`.".
  - Use markdown where necessary, mostly for `code blocks`.
  - End with either a period (.) or an exclamation mark (!).
  - Start with a capital letter.
* [ ] Pull request includes a [sign off](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#sign-off)
* [ ] Code style is correct (run the [linters](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#code-style))

### CODE REVIEW

**Comment by clokep** [comment]
@andrewperry Could you please sign off on your contribution (https://github.com/matrix-org/synapse/blob/develop/CONTRIBUTING.md#sign-off) and take a look at the CI failures? Thank you!

**Comment by anoadragon453** [comment]
The `mypy` failures are unrelated to these code changes and were an issue with the master branch a couple days ago. If you pull in the latest changes from master they should be fixed.

You do need to provide a changelog file though: https://github.com/matrix-org/synapse/blob/develop/CONTRIBUTING.md#changelog (or we can for you after you've signed off) :slightly_smiling_face:

File: docs/openid.md [hunk 1]
```diff
@@ -47,6 +47,15 @@ setting in your configuration file (or uncomment one of the existing examples).
...
 the text below for example configurations for specific providers.
 
+One thing to look out for if you are using a generic OpenID Connect provider is to
+manually set the config to use
+```yaml
+user_profile_method: userinfo_endpoint
+```
+as including the `openid` claim does NOT cause the default `auto` config to reliably
+use the userinfo endpoint, contrary to the homeserver.yaml comments. This may cause 
+the `preferred_username` to not be available when setting the localpart_template.
```
Comment (anoadragon453) [inline docs/openid.md:57]: This may be more a case of updating the sample config comments instead?

Comment (anoadragon453) [reply docs/openid.md:57]: Specifically, I think adding an explanation for something to look out for when setting `user_profile_method` would probably be better put in the config file which contains a description for each option. See https://github.com/matrix-org/synapse/blob/4bbd53545041c9295fbae5dd0cefdd66d55d0b53/synapse/config/oidc_config.py#L150-L155

**Comment by richvdh** [comment]
this seems to be abandoned, so I'm going to close it.

**Pull request closed without merging**